src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -780,6 +781,42 @@ public static bool IsValidFilePath([NotNullWhen(true)] string? fullPath)
         public static string NormalizeWithForwardSlash(string p)
             => DirectorySeparatorChar == '/' ? p : p.Replace(DirectorySeparatorChar, '/');
 
+        /// <summary>
+        /// Replaces all sequences of '\' or '/' with a single '/' but preserves UNC prefix '//'.
+        /// </summary>
+        public static string CollapseWithForwardSlash(string p)
```

Comment:
```suggestion
        public static string CollapseWithForwardSlash(ReadOnlySpan<char> p)
```

Consider using a `ReadOnlySpan<char>`. Makes it easier for future changes that are span related to call this API.